pkg/processor/instance.go

```diff
@@ -25,29 +24,27 @@ import (
 	"github.com/conduitio/conduit/pkg/record"
 )
 
-const (
-	TypeTransform Type = iota + 1
-	TypeFilter
-)
-
 const (
 	ParentTypeConnector ParentType = iota + 1
 	ParentTypePipeline
 )
 
-// Type defines the processor type.
-type Type int
-
 // ParentType defines the parent type of a processor.
 type ParentType int
 
-// Processor is the interface that represents a single message processor that
+// Interface is the interface that represents a single message processor that
 // can be executed on one record and manipulate it.
-type Processor interface {
-	// Type returns the processor type.
-	Type() Type
-	// Execute runs the processor function on a record.
-	Execute(ctx context.Context, record record.Record) (record.Record, error)
+type Interface interface {
+	// Process runs the processor function on a record.
+	Process(ctx context.Context, record record.Record) (record.Record, error)
+}
+
+// InterfaceFunc is a stateless processor.Interface implementation
```

Comment:
Nitpick of the highest order: it isn't _necessarily_ stateless 🙈
```
func NewStatefulProcessorFunc() processor.InterfaceFunc {
	myState := 0
	return func(_ context.Context, r record.Record) (record.Record, error) {
		myState++
		return r, nil
	}
}
```

Reply:
Yeah, it's an outdated comment from the version where I had a function field, so the old `funcProcessor` never stored any state itself, but was always just forwarding the calls to the fn function field. I'll update the comment. Thanks for pointing it out!